File: Python/_303_Range_Sum_Query/solution.py
class NumArray:

    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.currentSum = []
        self.currentSum.append(0)
        for i in range(0, len(nums)):
            self.currentSum.append(self.currentSum[i] + nums[i])

    def sumRange(self, i, j):
        """
        :type i: int
        :type j: int
        :rtype: int
        """
        return self.currentSum[j + 1] - self.currentSum[i];

# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)

File: Java/_303_Range_Sum_Query/Solution.java
class NumArray {
    
    private int[] currentSum;

    public NumArray(int[] nums) {
        currentSum = new int[nums.length + 1];
        currentSum[0] = 0;
        for (int i = 1; i < nums.length + 1; i++)
            currentSum[i] = currentSum[i - 1] + nums[i - 1];     
    }
    
    public int sumRange(int i, int j) {
        return currentSum[j + 1] - currentSum[i];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(i,j);
 */